qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1339,8 +1346,15 @@ def edge_weight_map(wire):
                     label=old_node.op.label,
                 )
             elif getattr(old_node.op, "condition", None) is not None:
-                m_op = copy.copy(old_node.op)
-                m_op.condition = variable_mapper.map_condition(m_op.condition)
+                # Deepcopy needed here in case of singletone gate usage the condition will be sticky
+                # globally
+                m_op = copy.deepcopy(old_node.op)
```

Comment:
This might be out-of-date, and if we've entered this path, then `old_node.op` _can't_ be singleton, right?

Reply:
I think you're right, I'll try removing it and see what happens.

Reply:
Removed in: https://github.com/Qiskit/qiskit/pull/10314/commits/b90eebbdf30caf0efcb2ddb5763e060e456c458a